clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1047,6 +1075,28 @@ class SyclKernelFieldHandlerBase {
   // Most handlers shouldn't be handling this, just the field checker.
   virtual bool handleOtherType(FieldDecl *, QualType) { return true; }
 
+  // Handle a simple struct that doesn't need to be decomposed, only called on
+  // handlers with VisitInsideSimpleContainers as false.  Replaces
+  // handleStructType, enterStruct, leaveStruct, and visiting of sub-elements.
+  virtual bool handleNonDecompStruct(const CXXRecordDecl *, FieldDecl *,
+                                     QualType) {
+    // TODO: Should this assert? Empty would have to do something about it.
```

Comment:
:D  Removed it (and the below).  I was considering making these assert to make sure I got all of the functions implemented when I cared (since you'd likely NEED to implement for all you've enabled this on), but it ended up not being an issue to make sure it is the case.